Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs

```diff
@@ -104,9 +97,30 @@ public void JoinWithIP(string ip, string port)
 
             m_GameNetPortal.PlayerName = m_PlayerNameLabel.text;
 
+            m_SignInSpinner.SetActive(true);
+
             m_ClientNetPortal.StartClient(ip, portNum);
```

Comment:
Shouldn't we do something similar here and check the return value of StartClient?

Reply:
If this returns false, UTP will retry in the background, following the reconnect attempts / connect timeout settings set on the UTP component. So, it's not enough to catch it just here.